### DISCUSSION TITLE
Quantity over time

### DISCUSSION THREAD

**Discussion by coupster74** [laurens94/vue-timeline-chart#3] [Q&A]
Hello! I love your control. We are considering using it on a project of ours, but I wanted to describe our use case and see how difficult it would be to augment/enhance the solution to meet our needs. 

We want to be able to visualize quantity as events part of the timeline, so if you are at the hourly level, you might see events throughout the day represented by points (as you have now), but as we zoom out to higher time buckets (day, days, week, weeks, month, months, etc), we want to see some visual representation of quantity for the time bucket. So while at the hour level, there might be 15 points, once the display shows at the day level, there would something to represent the quantity of 15. 

my original through was either size of dot to represent volume (but this could get messy as they overlay, and hard to understand and visualize growth), or a line graph... below is an example at the week bucket level with 3 series.

<img width="982" alt="image" src="https://github.com/laurens94/vue-timeline-chart/assets/6988322/7afe6f90-29de-4850-a1e8-eb15b752daa8">

Ideas? thoughts?

**Comment by laurens94** [comment]
Hi! I'm glad that you like it. :)

Interesting use case! 
I would suggest keeping track of the scale-level (I just added another event that you can use: `changeScale`) so you can serve the timeline a different set of events depending on the visible scale (day/hour/etc).

Then you could display the set of grouped-as-quantity events as ranges (as they never overlap and actually have a defined start and end time) and you can add custom styling to display them in various heights:

```css
:deep(.item.range) {
    height: var(--_height, 100%);
    bottom: 0;
    top: auto;
}
```

You can set the `--_height` variable for each event using `cssVariables`

So it might look similar to this (this is a screenshot from another project that uses this timeline):

![image](https://github.com/laurens94/vue-timeline-chart/assets/5780704/e27c72cc-2c1b-4e93-b64a-d37edff5a5a3)

It makes sense to add a tooltip to these range-events with the exact amount.

Let me know if this works for you! 👍

**Comment by laurens94** [comment]
@coupster74 I just added another slot (`items-$GROUPID`), so it's now possible to replace the entire plotting of items with some custom chart plotting library (such as [d3.js](https://github.com/d3/d3)). Maybe this is useful to you.

There's an example in App.vue, where you can see how this could be integrated.

<img width="779" alt="image" src="https://github.com/laurens94/vue-timeline-chart/assets/5780704/fcd9b24a-e905-4634-906c-3bc35f717468">

**Comment by coupster74** [comment]
That’s awesome. Thank you!
